https://issues.redhat.com/browse/ACM-13050 (#6744)
open-cluster-management/rhacm-docs: troubleshooting/acm_thanos_compactor.adoc

You might receive a block error message that indicates that the block for Thanos compactor is corrupted.

[#symptom-thanos-compactor]
== Symptom: Block error for Thanos compactor

After you upgrade {product-title} from 2.8 to 2.9 and check the logs for the Thanos compactor by using the `oc logs observability-thanos-compact-0` command, the logs display the following error message:

----
ts=2024-01-24T15:34:51.948653839Z caller=compact.go:491 level=error msg="critical error detected; halting" err="compaction: group 0@15699422364132557315: compact blocks [/var/thanos/compact/compact/0@15699422364132557315/01HKZGQGJCKQWF3XMA8EXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HKZQK7TD06J2XWGR5EXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HKZYEZ2DVDQXF1STVEXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HM05APAHXBQSNC0N5EXAMPLE]: populate block: chunk iter: cannot populate chunk 8 from block 01HKZYEZ2DVDQXF1STVEXAMPLE: segment index 0 out of range"
----

[#resolving-thanos-compactor]
== Resolving the problem: Add the _thanos bucket verify_ command

If you have this problem, complete the following steps:

> . Resolve the block error by adding the `thanos bucket verify` command to the object storage configuration. Set the configuration in the `observability-thanos-compact` pod by using the following command:

Author: ```suggestion
. Resolve the block error by adding the `thanos bucket verify` command to the object storage configuration. Set the configuration in the `observability-thanos-compact` pod by using the following commands:
```

> //I'm not sure what [..] represents and should there be multiple commands because the hashtag seems like its a hidden comment

Reviewer: I think `[...]` is just an ellipsis to avoid having to specify the command's output.

> //I'm not sure what [..] represents and should there be multiple commands because the hashtag seems like its a hidden comment

Author: ```suggestion
```

+
[source,bash]
----
oc rsh observability-thanos-compact-0
[..]

> # thanos tools bucket verify -r --objstore.config="$OBJSTORE_CONFIG" --objstore-backup.config="$OBJSTORE_CONFIG" --id=01HKZYEZ2DVDQXF1STVEXAMPLE

Reviewer: I think this hashtag can be removed. I don't see it being important for anything in this context.

> # thanos tools bucket verify -r --objstore.config="$OBJSTORE_CONFIG" --objstore-backup.config="$OBJSTORE_CONFIG" --id=01HKZYEZ2DVDQXF1STVEXAMPLE

Author: ```suggestion
thanos tools bucket verify -r --objstore.config="$OBJSTORE_CONFIG" --objstore-backup.config="$OBJSTORE_CONFIG" --id=01HKZYEZ2DVDQXF1STVEXAMPLE
```

[#troubleshooting-thanos-compactor]
= Troubleshooting a block error for Thanos compactor

You might receive a block error message that indicates that the block for Thanos compactor is corrupted.

[#symptom-thanos-compactor]
== Symptom: Block error for Thanos compactor

After you upgrade {product-title} from 2.8 to 2.9 and check the logs for the Thanos compactor by using the `oc logs observability-thanos-compact-0` command, the logs display the following error message:

----
ts=2024-01-24T15:34:51.948653839Z caller=compact.go:491 level=error msg="critical error detected; halting" err="compaction: group 0@15699422364132557315: compact blocks [/var/thanos/compact/compact/0@15699422364132557315/01HKZGQGJCKQWF3XMA8EXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HKZQK7TD06J2XWGR5EXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HKZYEZ2DVDQXF1STVEXAMPLE /var/thanos/compact/compact/0@15699422364132557315/01HM05APAHXBQSNC0N5EXAMPLE]: populate block: chunk iter: cannot populate chunk 8 from block 01HKZYEZ2DVDQXF1STVEXAMPLE: segment index 0 out of range"
----

[#resolving-thanos-compactor]
== Resolving the problem: Add the _thanos bucket verify_ command

> If you have this problem, complete the following steps:

Reviewer 2: They already have the problem, right? So we can remove _If you have this problem,_.

Since we do need a lede:

Something like:

Add the `thanos bucket verify` command to the object storage configuration. Complete.....

> If you have this problem, complete the following steps:

Author: ```suggestion
Add the `thanos bucket verify` command to the object storage configuration. Complete the following steps:
```

. Resolve the block error by adding the `thanos bucket verify` command to the object storage configuration. Set the configuration in the `observability-thanos-compact` pod by using the following commands:

+
[source,bash]
----
oc rsh observability-thanos-compact-0
[..]
thanos tools bucket verify -r --objstore.config="$OBJSTORE_CONFIG" --objstore-backup.config="$OBJSTORE_CONFIG" --id=01HKZYEZ2DVDQXF1STVEXAMPLE
----

. If the previous command does not work, you must mark the block for deletion because it might be corrupted. Run the following commands:

+
[source,bash]
----
thanos tools bucket mark --id "01HKZYEZ2DVDQXF1STVEXAMPLE" --objstore.config="$OBJSTORE_CONFIG" --marker=deletion-mark.json --details=DELETE 
----

> .. Clean up the marked blocks by running the following command:

Reviewer 2: This is good, but not sure it would be a sub step, since you only have one. Not gonna nit pick about it, but it could be the next step, but lead with:

. If you blocked for deletion, clean up.....

> .. Clean up the marked blocks by running the following command:

Author: ```suggestion
. If you blocked for deletion, clean up the marked blocks by running the following command:
```
